### PR TITLE
Nirs actionnetwork changes

### PR DESCRIPTION
**Pull request by NirTatcher**
Hi @shaunagm, Round 2 of functions (POST, PUT and even DELTE just for taggings) to create or update any resource that is available to be updated.
Please let me know if there is any feedback!
Thanks

### CODE REVIEW

File: parsons/action_network/action_network.py [hunk 1]
```diff
@@ -169,6 +169,32 @@ def get_person_attendance(self, person_id, attendance_id):
...
 
+    def create_attendance(self, event_id, payload):
+        """
+        `Args:`
+            event_id: the unique id of the event
+            payload: the payload for creating the event attendance
+        `Returns:`
+            A JSON response after creating the event attendance
+        """
+        return self.api.post_request(f"events/{event_id}/attendances", payload)
```
Comment (shaunagm) [inline parsons/action_network/action_network.py:180]: I'm noticing now that a lot of these methods just say "the payload". I imagine that the payloads for a lot of these items are very detailed and we don't want to put all that in the docstrings, but perhaps we could link to the payload details? For example this one could link to https://actionnetwork.org/docs/v2/attendances

File: parsons/action_network/action_network.py [hunk 2]
```diff
@@ -1139,6 +1412,34 @@ def get_person_signature(self, person_id, signature_id):
...
+            petition_id:
+               The unique id of the petition
+            data:
+               The data for creating the signature
+        `Returns:`
+            A JSON with the created signature entry
+        """
+        return self.api.post_request(f"petitions/{petition_id}/signatures", data)
+
+    def update_signature(self, petition_id, signature_id, data):
```
Comment (shaunagm) [inline parsons/action_network/action_network.py:1427]: Why is this one "data" and not "payload"?

File: test/test_action_network/test_action_network.py [hunk 3]
```diff
@@ -3874,6 +3985,39 @@ def test_get_message(self, m):
             self.fake_message,
         )
 
+    @requests_mock.Mocker()
+    def test_create_message(self, m):
+        payload = {
+            "subject": "Stop doing the bad thing",
```
Comment (shaunagm) [inline test/test_action_network/test_action_network.py:3991]: 😂 I love test data

**Comment by shaunagm** [comment]
@NirTatcher this looks great overall, my only concern is about having (a) consistency and (b) enough details in the parameters to the various methods.

**Review by shaunagm — CHANGES_REQUESTED** [review]

File: parsons/action_network/action_network.py (hunk 1 above)
Comment (NirTatcher) [reply parsons/action_network/action_network.py:200]: Hi @shaunagm! 
I hope that is ok I've added a "Documentation Reference" for each function for example:
        """
        `Args:`
            tag_id:
                The unique id of the tag
            payload:
                The payload for creating the tagging
                {
                    "_links" : {
                        "osdi:person" : { "href" : "https://actionnetwork.org/api/v2/people/id" }
                    }
                }
        `Returns:`
            A JSON response after creating the tagging
        **`Documentation Reference`:
            https://actionnetwork.org/docs/v2/taggings**
        """
Please let me know if this is acceptable or if you prefer another way to attach a reference from the documentation!

File: parsons/action_network/action_network.py (hunk 2 above)
Comment (NirTatcher) [reply parsons/action_network/action_network.py:1659]: You're right, didn't see it! I changed it to payload :)

File: parsons/action_network/action_network.py (hunk 1 above)
Comment (shaunagm) [reply parsons/action_network/action_network.py:200]: This is perfect, thanks!

**Review by shaunagm — APPROVED** [review]
Thanks for making the changes, these look great! Apologies for the delay - hope you enjoyed the holidays :)

**Pull request merged**